core/src/saros/account/XMPPAccountStore.java

```diff
@@ -150,44 +149,21 @@ private synchronized void loadAccounts() {
 
     LOG.debug("loading accounts from file: " + accountFile.getAbsolutePath());
 
-    DataInputStream dataIn = null;
-
-    int size;
-
-    byte[] buffer;
+    FileInputStream dataIn = null;
 
     try {
-      dataIn = new DataInputStream(new FileInputStream(accountFile));
-
-      size = dataIn.readInt();
-
-      if (size <= 0 || size > MAX_ACCOUNT_DATA_SIZE)
-        throw new IOException(
-            "account data seems malformed, refused to load " + size + " bytes into memory");
-
-      buffer = new byte[size];
-
-      dataIn.readFully(buffer);
-
-      activeAccount =
-          (XMPPAccount)
-              new ObjectInputStream(new ByteArrayInputStream(Crypto.decrypt(buffer, secretKey)))
-                  .readObject();
-
-      size = dataIn.readInt();
+      dataIn = new FileInputStream(accountFile);
 
-      if (size <= 0 || size > MAX_ACCOUNT_DATA_SIZE)
-        throw new IOException(
-            "account data seems malformed, refused to load " + size + " bytes into memory");
+      byte[] encryptedAccountData = IOUtils.toByteArray(dataIn);
```

Comment:
So what happens if you point to a garbage file with about several 100 MBytes in size ?

Reply:
I will add a check whether the file is smaller than MAX_ACCOUNT_DATA_SIZE

Reply:
Done